docs/architecture/data.md

```diff
@@ -111,3 +113,46 @@ class first_gcs,second_gcs,validated_gcs gcs_group_boxstyle
 class first_bq,second_bq bq_group_boxstyle
 class gcs_label1,gcs_label2,gcs_label3,bq_label1,bq_label2 group_labelstyle
 ```
+
+## Adding a new data source
+
+Adding a new data source based on the architecture described above involves several steps, outlined below.
+
+```{note}
+If you're bringing in data that is similar to existing data (for example, a new subset of an existing dataset like a new Airtable or Littlepay table), you should follow the existing pattern for that dataset. [This spreadsheet](https://docs.google.com/spreadsheets/d/1bv1K5lZMnq1eCSZRy3sPd3MgbdyghrMl4u8HvjNjWPw/edit#gid=0) gives overviews of some prominent existing data sources, outlining the specific code/resources that correspond to each step in the [general data flow](data-ingest-diagram) for that data source.
+```
+
+### Determine upstream source type
+
+To determine the best storage location for your raw data (especially if it requires manual curation), consult the [Data Collection and Storage Guidance within the Cal-ITP Data Pipeline Google Doc](https://docs.google.com/document/d/1-l6c99UUZ0o3Ln9S_CAt7iitGHvriewWhKDftESE2Dw/edit).
+
+The [Should it be a dbt model?](tool_choice) docs section also has some guidance about when a data pipeline should be created.
+
+### Bring data into Google Cloud Storage
+
+We store our raw, un-transformed data in Google Cloud Storage to ensure that we can always recover the raw data if needed.
+
+We store data in [hive-partitioned buckets](https://cloud.google.com/bigquery/docs/hive-partitioned-queries#supported_data_layouts) so that data is clearly labeled and partitioned for better performance. We use UTC dates and timestamps in hive paths (for example, for the timestamp of the data extract) for consistency.
+
+You will need to set up a way to bring your raw data into the Cal-ITP Google Cloud Storage environment. Most commonly, we use [Airflow](https://airflow.apache.org/) for this.
+
+The [Airflow README in the data-infra repo](https://github.com/cal-itp/data-infra/tree/main/airflow#readme) has information about how to set up Airflow locally for testing and how the Airflow project is structured.
+
+We often bring data into our environment in two steps, created as two separate Airflow [DAGs](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html):
+
+- **Sync the fully-raw data in its original format:** See for example the changes in the `airflow/dags/sync_elavon` directory in [data-infra PR #2376](https://github.com/cal-itp/data-infra/pull/2376/files). We do this to preserve the raw data in its original form. This data might be saved in a `calitp-<your-data-source>-raw` bucket.
```

Comment:
Elavon may not be the best example here since we sync the world each day, which is generally not what we do for other pipelines. But if there isn't another clean-cut example that adheres to most of our other patterns, I'm more than willing to stick with this one.

Reply:
After checking, there isn't actually an Airflow-managed pipeline that filters a request by date or similar -- most just scrape whatever they find at the time of search, so I actually think Elavon is not unusual in this regard. I am adding a little clarification here though